src/core/lib/iomgr/tcp_posix.cc

```diff
@@ -606,9 +608,11 @@ static bool tcp_flush(grpc_exec_ctx* exec_ctx, grpc_tcp* tcp,
 
     if (tcp->outgoing_slice_idx == tcp->outgoing_buffer->count) {
       *error = GRPC_ERROR_NONE;
+      grpc_slice_buffer_reset_and_unref_internal(exec_ctx,
+                                                 tcp->outgoing_buffer);
```

Comment:
If you always unref and "remove" the slice as you send out, tcp->outgoing_slice_idx is going to be always 0 and you may want to use a local variable to replace that in this function.

Reply:
Done